### PR TITLE
composefile-smb: Added docker compose file for smb.

### PR DESCRIPTION
**Pull request by grizmin**

### CODE REVIEW

File: timemachine-compose-afp.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - SET_PERMISSIONS=false
+      - SHARE_NAME=TimeMachine
+      - VOLUME_SIZE_LIMIT=0
+    restart: unless-stopped
+    ports:
+     - "548:548"
+     - "636:636"
+    volumes:
+     - /path/to/your/timemachine/volume:/opt/timemachine
+     - ./timemachine-netatalk:/var/netatalk
```
Comment (mbentley) [inline timemachine-compose-afp.yml]: Same comment on the volumes here with a relative path vs a docker managed volume.

File: timemachine-compose-smb.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      - SHARE_NAME=TimeMachine
+      - VOLUME_SIZE_LIMIT=0
+      - WORKGROUP=WORKGROUP
+    restart: unless-stopped
+    ports:
+     - "548:548"
+     - "636:636"
+    volumes:
+     - /path/to/your/timemachine/volume:/opt/timemachine
+     - ./timemachine-var-log:/var/log
```
[the remark this answers is not in the document]
Comment (grizmin) [reply timemachine-compose-smb.yml]: This is how I run it @home. Not sure if this is the right paradigm or not. IMO the docker compose will be used by people who want to test the solution. If they like it, they might spend time to read the whole readme and implement their own changes. I even thought if not making the backup path  relative in order top make the compose file usable without any editing.

File: timemachine-compose-smb.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - TM_UID=1000
+      - TM_GID=1000
+      - PASSWORD=timemachine
+      - SET_PERMISSIONS=false
+      - SHARE_NAME=TimeMachine
+      - VOLUME_SIZE_LIMIT=0
+      - WORKGROUP=WORKGROUP
+    restart: unless-stopped
+    ports:
+     - "548:548"
```
[the remark this answers is not in the document]
Comment (grizmin) [reply timemachine-compose-smb.yml]: True, I kept them for verbosity if someone want's to use different kind of networking.
+ few bytes, right ? :)

File: timemachine-compose-afp.yml (hunk 1 above)
Comment (grizmin) [reply timemachine-compose-afp.yml]: same answer

File: timemachine-compose-smb.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbentley) [reply timemachine-compose-smb.yml]: The ports are not correct for the SMB example, they should be:
137:137/udp
138:138/udp
139:139

not: 
548:548
636:636

As 548 and 636 are used for AFP.

**Pull request merged**